### PR TITLE
curl error msg for low version

### PR DESCRIPTION
**Pull request by wwulfric**
lower version curl wouldn't tell uri in error message.

### CODE REVIEW

**Comment by sagikazarmark** [comment]
Newer versions does, so this solution sort of duplicates the uri in the error message? Maybe format the message according to the curl version?

**Comment by wwulfric** [comment]
@sagikazarmark Solved

File: src/Handler/CurlFactory.php [hunk 1]
```diff
@@ -14,6 +14,8 @@
  */
 class CurlFactory implements CurlFactoryInterface
 {
+    private const CURL_VERSION_STR = 'curl_version';
```
Comment (sagikazarmark) [inline src/Handler/CurlFactory.php]: Constant visibility is not supported on PHP 5.x

**Comment by sagikazarmark** [comment]
Thanks @wwulfric

**Comment by wirwolf** [comment]
Hello @wwulfric, i create a fork of this repo and release this marge request into the new version(6.4.0).

Namespace compatibility is 100%, and you can use my changes if you patch composer.json in your projects
* replace `"guzzlehttp/guzzle": "^6.3"` to `"someblackmagic/guzzle": "^6.4"` 
* add `"guzzlehttp/guzzle": "6.3.*"` into replace block

Fork link: https://github.com/SomeBlackMagic/guzzle/

File: src/Handler/CurlFactory.php [hunk 2]
```diff
@@ -172,13 +176,22 @@ private static function createRejection(EasyHandle $easy, array $ctx)
...
             );
         }
-
-        $message = sprintf(
-            'cURL error %s: %s (%s)',
-            $ctx['errno'],
-            $ctx['error'],
-            'see https://curl.haxx.se/libcurl/c/libcurl-errors.html'
-        );
+        if (version_compare($ctx[self::CURL_VERSION_STR], self::LOW_CURL_VERSION_NUMBER)) {
```
Comment (guilliamxavier) [inline src/Handler/CurlFactory.php:179]: Sorry to be late but this seems wrong, it's missing a comparison operator (either against 0 or as a third string argument, see [examples](https://www.php.net/manual/en/function.version-compare.php#refsect1-function.version-compare-examples)). Currently it's basically testing `curl_version()['version'] != '7.21.2'`. I guess you intended `>=` here. But actually I couldn't find any evidence that curl 7.21.2+ added the URI in the error message... Could you please explain your scenario?

Comment (guilliamxavier) [reply src/Handler/CurlFactory.php:179]: ping @sagikazarmark: the code needs to be fixed!
Either you have evidence that curl 7.21.2+ added the URI in the error message, and then you should add `>= 0` (or `> 0`) between `))`;
or you don't, and then this should be reverted, and maybe re-submit only https://github.com/guzzle/guzzle/pull/2108/commits/bd368857d54081a9eeaea3c0ecfcb4e10455e56b

Comment (guilliamxavier) [reply src/Handler/CurlFactory.php:179]: proposed #2648

**Pull request merged**